install/management/appliance.go

```diff
@@ -342,18 +324,24 @@ func (d *Dispatcher) setMacToGuestInfo() error {
 		ExtraConfig: []types.BaseOptionValue{},
 	}
 	var keys []string
-	for key, value := range m {
-		if value == "" {
+
+	for nicName, netInfo := range conf.Networks {
+		mac, ok := m[netInfo.PortGroupName]
+		if !ok || mac == "" {
 			// timeout to wait MAC address, so empty mac address is returned
 			err = errors.Errorf("Timeout to get VM MAC address")
 			return err
 		}
-		netName := d.nics[key]
-		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s/portgroup", netName), Value: key})
-		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s/mac", netName), Value: value})
-		keys = append(keys, netName)
+
+		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s/portgroup", nicName), Value: netInfo.PortGroupName})
```

Comment:
This is already in VCH config, https://github.com/vmware/vic/pull/730/files#diff-f8dda6a8e934bfbbf6e19f4868adef44R110
Just I think you'll remove the old code in launcher.sh, I just keep the existing format here, instead of change launcher.sh to read new guestinfo format.
So while you change to read VirtualContainerHostConfigSpec.Networks, these configuration should be removed as well.